core/src/main/scala/akka/kafka/javadsl/Producer.scala

```diff
@@ -73,6 +73,26 @@ object Producer {
       .mapMaterializedValue(_.toJava)
       .asJava
 
+  /**
+   * Sink that is aware of the [[ConsumerMessage.TransactionalMessage#PartitionOffset]] from a [[Consumer]].  It will
+   * initialize, begin, produce, and commit the consumer offset as part of a transaction.
+   */
+  def transactionalSink[K, V](
+    settings: ProducerSettings[K, V],
+    transactionalId: String
+  ): Sink[Message[K, V, ConsumerMessage.PartitionOffset], CompletionStage[Done]] =
```

Comment:
There are 2 other common use cases - for single input message:
- someone would like to do the message processing and only commit the input message, so there would be no output kafka message
- someone would like to emit multiple messages (possible for different output topics)

This could be modeled as `Sink[Seq[Message[K, V, ConsumerMessage.PartitionOffset]], CompletionStage[Done]]`

Reply:
Interesting.  I didn't consider this use case.  I'll think about how to support it.

Reply:
@rafalmag Couldn't this use case be satisfied with non-Transactional consumer group offset commits after message is processed?  What benefit does the user get when they actually use a transaction?